fluent-bundle/src/resolve.rs

```diff
@@ -154,9 +155,12 @@ impl<'source> ResolveValue for ast::Expression<'source> {
                                 }
                             }
                             ast::VariantKey::NumberLiteral { value } => {
-                                let key = FluentValue::as_number(value).unwrap();
-                                if key.matches(env.bundle, selector) {
-                                    return variant.value.to_value(env);
+                                if let Ok(key) = FluentValue::as_number(value) {
+                                    if key.matches(env.bundle, selector) {
+                                        return variant.value.to_value(env);
+                                    }
+                                } else {
+                                    return Err(ResolverError::Value);
```

Comment:
`map_or_else`?

Reply:
That's nightly-only, according to clippy. Also, we need the early return to exit the `for` loop but only when the `key` matches. Can you do that with a closure?

Reply:
nope, all good!